storage/storage_image.go

```diff
@@ -577,7 +577,7 @@ func (s *storageImageDestination) Commit(ctx context.Context) error {
 		// Build the new layer using the diff, regardless of where it came from.
 		// TODO: This can take quite some time, and should ideally be cancellable using ctx.Done().
 		layer, _, err := s.imageRef.transport.store.PutLayer(id, lastLayer, nil, "", false, nil, diff)
-		if err != nil {
+		if err != nil && errors.Cause(err) != storage.ErrDuplicateID {
 			return errors.Wrapf(err, "error adding layer with blob %q", blob.Digest)
 		}
 		lastLayer = layer.ID
```

Comment:
We've seen a panic on Azure because of this change. If err == ErrDuplicateID, the layer object is nil and causes a nil pointer dereference here. @nalind @rhatdan PTAL

Reply:
created https://github.com/containers/storage/pull/234

Reply:
See also https://github.com/containers/libpod/issues/1730 .

Reply:
https://github.com/containers/storage/pull/234